tests/system/python/e2e/test_e2e_notification_service_with_plugins.py

```diff
@@ -0,0 +1,214 @@
+    assert SERVICE_NAME == audit_detail['details']['name']
+
+
+class TestNotificationService:
+
+    def test_service(self, reset_and_start_foglamp, service_branch, foglamp_url, wait_time, retries, remove_directories):
+        _configure_and_start_service(service_branch, foglamp_url, remove_directories)
+
+        retry_count = 0
+        # only 2 services is being up by default i.e core and storage
+        default_registry_count = 2
+        service_registry = default_registry_count
+        while service_registry != 3 and retry_count < retries:
+            svc = _get_result(foglamp_url, '/foglamp/service')
+            service_registry = svc['services']
+            retry_count += 1
+            time.sleep(wait_time * 2)
+
+        if len(service_registry) == default_registry_count:
+            assert False, "Failed to start the {} service".format(SERVICE)
+
+        _verify_service(foglamp_url, 'running')
+
+        _verify_audit_log_entry(foglamp_url, '/foglamp/audit?source=NTFST')
+
+    def test_get_default_notification_plugins(self, foglamp_url, remove_directories):
+        remove_directories(os.environ['FOGLAMP_ROOT'] + '/plugins/notificationDelivery')
+        remove_directories(os.environ['FOGLAMP_ROOT'] + 'cmake_build/C/plugins/notificationDelivery')
+        jdoc = _get_result(foglamp_url, '/foglamp/notification/plugin')
+        assert [] == jdoc['delivery']
+        assert 1 == len(jdoc['rules'])
+        assert NOTIFY_RULE == jdoc['rules'][0]['name']
+
+
+class TestNotificationCRUDs:
+
+    # FIXME: FOGL-2434 Add name with some special character
+    @pytest.mark.parametrize("data", [
+        {"name": "Test1", "description": "Test 1 notification", "rule": NOTIFY_RULE, "channel": NOTIFY_PLUGIN, "enabled": "false", "notification_type": "retriggered"},
+        {"name": "Test2", "description": "Test 2 notification", "rule": NOTIFY_RULE, "channel": NOTIFY_PLUGIN, "enabled": "false", "notification_type": "toggled"},
+        {"name": "Test3", "description": "Test 3 notification", "rule": NOTIFY_RULE, "channel": NOTIFY_PLUGIN, "enabled": "false", "notification_type": "one shot"}
+    ])
+    def test_create_notification_instances_with_default_rule_and_channel_python35(self, foglamp_url, notify_branch, data, remove_directories):
+        def _add_notification_instance(payload):
+            conn = http.client.HTTPConnection(foglamp_url)
+            conn.request("POST", '/foglamp/notification', json.dumps(payload))
+            r = conn.getresponse()
+            assert 200 == r.status
+            r = r.read().decode()
+            jdoc = json.loads(r)
+            assert "Notification {} created successfully".format(payload['name']) == jdoc['result']
+
+        # FIXME: Handle in a better way; we need below code once for a test
+        if data['name'] == 'Test1':
+            _install_notify_plugin(notify_branch, NOTIFY_PLUGIN, remove_directories)
+
+        _add_notification_instance(data)
```

Comment:
how about the cleanup of these?

Reply:
For system tests we are not cleaning any of stuff inside foglamp instance...as `foglamp reset` will do on next run